[BUG] github tests: build-linux ptl fails to heap_uncached_part failure

**Describe the bug**
Regression in commit 9cc7ad94d1e378c3a2b128178330321d99787dc8

https://github.com/thesofproject/sof/actions/runs/20127912805/job/57761888407?pr=10429

```
 /__w/sof/sof/sof/src/audio/module_adapter/library/userspace_proxy.c: In function 'userspace_proxy_memory_init':
/__w/sof/sof/sof/src/audio/module_adapter/library/userspace_proxy.c:77:18: error: 'heap_part_cached' undeclared (first use in this function)
   77 |                 &heap_part_cached,
      |                  ^~~~~~~~~~~~~~~~
```

Problem uncovered when USERSPACE_PROXY was enabled in a default build target.

--- a/src/audio/module_adapter/library/userspace_proxy.c
+++ b/src/audio/module_adapter/library/userspace_proxy.c
@@ -74,7 +74,7 @@ static int userspace_proxy_memory_init(struct userspace_context *user,
 
 	struct k_mem_partition *parts_ptr[] = {
 #ifdef HEAP_PART_CACHED
-		&heap_part_cached,
+		&heap_cached_part,
 #endif
 		&heap_part
 	};

